src/vcCamera.cpp

```diff
@@ -194,26 +194,31 @@ void vcCamera_Apply(vcState *pProgramState, vcCamera *pCamera, vcCameraSettings
     if (distanceToPointSqr != 0.0 && (pCamInput->mouseInput.x != 0 || pCamInput->mouseInput.y != 0))
     {
       // Orbit Left/Right
-      udDoubleQuat rotation = udDoubleQuat::create(worldAnchorNormal, pCamInput->mouseInput.x);
-      udDouble3 direction = pCamera->position - pProgramState->worldAnchorPoint; // find current direction relative to center
+      if (pCamInput->mouseInput.x != 0)
+      {
+        udDoubleQuat rotation = udDoubleQuat::create(worldAnchorNormal, pCamInput->mouseInput.x);
+        udDouble3 direction = pCamera->position - pProgramState->worldAnchorPoint; // find current direction relative to center
 
-      orientation = (rotation * orientation);
+        orientation = (rotation * orientation);
 
-      pCamera->position = pProgramState->worldAnchorPoint + rotation.apply(direction); // define new position
-      pCamera->headingPitch = vcGIS_QuaternionToHeadingPitch(pProgramState->gis, pProgramState->camera.position, orientation);
+        pCamera->position = pProgramState->worldAnchorPoint + rotation.apply(direction); // define new position
+        pCamera->headingPitch = vcGIS_QuaternionToHeadingPitch(pProgramState->gis, pProgramState->camera.position, orientation);
+      }
+
+      //
+      if (pCamera->headingPitch.y < UD_DEG2RADf(-89.0) && pCamInput->mouseInput.y <= 0) break;
+      if (pCamera->headingPitch.y > UD_DEG2RADf(89.0) && pCamInput->mouseInput.y >= 0) break;
```

Comment:
the `break` should be on the next line (even if it is less visually appealing)

Reply:
fixed.